src/Feature/Changelog/Feature_Changelog.re

```diff
@@ -32,6 +43,25 @@ let isSameDate = (a, b) => {
   a.tm_year == b.tm_year && a.tm_yday == b.tm_yday;
 };
 
+let update = (model, msg) =>
+  switch (msg) {
+  | PullRequestClicked(pr) =>
+    let url = Printf.sprintf("https://github.com/onivim/oni2/pull/%d", pr);
+    (model, URL(url));
+  | CommitHashClicked(hash) =>
+    let url =
+      Printf.sprintf("https://github.com/onivim/oni2/commit/%s", hash);
+    (model, URL(url));
```

Comment:
These should really just emit effects directly, but that requires moving the `openURL` effect to some place that's accessible here. Perhaps you could move both this and `stat` to a `Service_OS` project? Or maybe we should have a `Service_Revery` project for this, and a `Service_Filesystem` for `stat`?

Reply:
Also, `URL` is not a great name for a `msg`, outgoing or otherwise, as it says nothing about either what has happened or what's expected to happen. If you later have a URL that you need a different behavior for, then now you have a problem (well, ambiguous code is already a problem of course, but this demonstrates why it's ambiguous).

Reply:
I changed this around in the latest commit to what I _think_ you mean. Let me know if I still need to change anything!

Reply:
Perfect. Thanks!